src/detect-engine-tag.c

```diff
@@ -784,35 +772,27 @@ static int DetectTagTestPacket02 (void)
             expect = false;
         else
             expect = true;
-        if (((p[i]->flags & PKT_HAS_TAG) ? true : false) != expect)
-            goto cleanup;
+        FAIL_IF(((p[i]->flags & PKT_HAS_TAG) ? true : false) != expect);
     }
 
-    result = 1;
-
-cleanup:
     UTHFreePackets(p, 7);
-    if (det_ctx != NULL)
-        DetectEngineThreadCtxDeinit(&th_v, (void *)det_ctx);
+    DetectEngineThreadCtxDeinit(&th_v, (void *)det_ctx);
+
+    SigGroupCleanup(de_ctx);
+    SigCleanSignatures(de_ctx);
```

Comment:
Should I remove `SigGroupCleanup` and `SigCleanSignatures`?

Reply:
ya, just `DetectEngineCtxFree` is enough